R-package/src/xgboost_R.cc

```diff
@@ -23,6 +23,31 @@
 #include "./xgboost_R.h"  // Must follow other includes.
 
 namespace {
+
+struct ErrorWithUnwind : public std::exception {};
+
+void throw_exception_from_R_error(void *unused, Rboolean jump) {
```

Comment:
That's the only documentation there is.

In case it helps: when a failure occurs, the long jump is triggered from the line that calls `R_ContinueUnwind`. By that point, all the C++ objects will have been destructed, since that line passes through the 'try' block. Upon throwing an R error, all the R allocations that happened during the `.Call` part are freed.

Simpler example of this usage can be found here: https://github.com/david-cortes/unwindprotect

Reply:
Thank you for the explanation and the example usage. This has been most helpful.